packages/webapp/src/containers/Task/TaskAssignment/index.jsx

```diff
@@ -19,58 +26,94 @@ export default function TaskManagement({ history, match, location }) {
   const users = userFarms[farm_id];
   const userData = Object.values(users).filter((user) => user.status !== 'Inactive');
   const persistedFormData = useSelector(hookFormPersistSelector);
-  const selectedTaskType = useSelector(taskTypeSelector(persistedFormData.task_type_id));
-  const isCustomType = !!selectedTaskType.farm_id;
-  const [options, setOptions] = useState([{ label: t('TASK.UNASSIGNED'), value: null }]);
-  const [wageData, setWageData] = useState([
-    { 0: { currency: null, hourly_wage: null, currencySymbol: null } },
-  ]);
   const [isFarmWorker] = useState(userFarm.role_id === 3);
-  const currencySymbol = grabCurrencySymbol(getCurrencyFromStore());
   const worker = users[userFarm.user_id];
 
-  useEffect(() => {
-    let wage_data = [];
-    let innerOptions = [];
-    if (userFarm.role_id === 3) {
-      let obj = { label: worker.first_name + ' ' + worker.last_name, value: worker.user_id };
-      let wageObj = {
-        [worker.user_id]: {
-          currency: worker.units.currency,
-          hourly_wage: worker.wage.amount,
-          currencySymbol: currencySymbol,
-        },
-      };
-      innerOptions.push(obj);
-      wage_data.push(wageObj);
-    } else {
-      for (let i = 0; i < userData.length; i++) {
-        let obj = {
-          label: userData[i].first_name + ' ' + userData[i].last_name,
-          value: userData[i].user_id,
-        };
-        innerOptions.push(obj);
-        let wageObj = {
-          [userData[i].user_id]: {
-            currency: userData[i].units.currency,
-            hourly_wage: userData[i].wage.amount,
-            currencySymbol: currencySymbol,
-          },
+  const defaultAssignee = useMemo(() => {
+    let { assignee } = persistedFormData;
+    if (!assignee) {
+      if (isFarmWorker || userData.length === 1) {
+        // "current user" if he/she is a farm worker or if he/she is the only user at the farm
+        assignee = {
+          label: worker.first_name + ' ' + worker.last_name,
+          value: worker.user_id,
         };
-        wage_data.push(wageObj);
+      } else {
+        // “Unassigned” if more than one user exists
+        assignee = { label: t('TASK.UNASSIGNED'), value: null };
       }
     }
-    setOptions(options.concat(innerOptions));
-    setWageData(wageData.concat(wage_data));
```

Comment:
`options` is now generated in `useTaskAssignForm` and user's wage is also provided from the hook.
`wageData` was a map `{[user_id]: { hourly_wage: xx, ... }}` only to get user's wage.